File: app/src/main/java/org/stepic/droid/model/Dataset.java
package org.stepic.droid.model;

import org.jetbrains.annotations.Nullable;

import java.util.List;

public class Dataset {
    private boolean is_multiple_choice;
    private List<String> options;
    private String someStringValueFromServer;
    private List<Pair> pairs;
    private Boolean is_html_enabled;
    private List<FillBlankComponent> components;
    private List<String> rows;
    private List<String> columns;
    private String description;
    private boolean is_checkbox;

    public Dataset() {
    }

    public Dataset(String someStringValueFromServer) {
        this.someStringValueFromServer = someStringValueFromServer;
    }

    public boolean is_multiple_choice() {
        return is_multiple_choice;
    }

    public List<String> getOptions() {
        return options;
    }

    @Nullable
    public String getSomeStringValueFromServer() {
        return someStringValueFromServer;
    }

    @Nullable
    public List<Pair> getPairs() {
        return pairs;
    }

    public Boolean getIs_html_enabled() {
        return is_html_enabled;
    }

    public List<FillBlankComponent> getFillBlankComponents() {
        return components;
    }

    public String getDescriptionTableQuiz() {
        return description;
    }

    public boolean isTableCheckbox() {
        return is_checkbox;
    }

    public List<String> getTableRows() {
        return rows;
    }

    public List<String> getTableColumns() {
        return columns;
    }
}

File: model/src/main/java/org/stepik/android/model/attempts/Dataset.kt
package org.stepik.android.model.attempts

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.android.parcel.Parcelize

@Parcelize
data class Dataset(
    val options: List<String>? = null,
    val someStringValueFromServer: String? = null,
    val pairs: List<Pair>? = null,
    val rows: List<String>? = null,
    val columns: List<String>? = null,
    val description: String? = null,
    val components: List<Component>? = null,

    @SerializedName("is_multiple_choice")
    val isMultipleChoice: Boolean = false,
    @SerializedName("is_checkbox")
    val isCheckbox: Boolean = false,
    @SerializedName("is_html_enabled")
    val isHtmlEnabled: Boolean = false
) : Parcelable

data class DatasetWrapper(
    val dataset: Dataset? = null
) : Parcelable {
    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeParcelable(dataset, flags)
    }

    override fun describeContents(): Int = 0

    companion object CREATOR : Parcelable.Creator<DatasetWrapper> {
        override fun createFromParcel(parcel: Parcel): DatasetWrapper =
            DatasetWrapper(parcel.readParcelable(Dataset::class.java.classLoader))

        override fun newArray(size: Int): Array<DatasetWrapper?> =
            arrayOfNulls(size)
    }
}